concepts/05 UI Components/DataGrid/15 Columns/40 Customize Cells/3 Customize Row Apperance.md

```diff
@@ -0,0 +1,147 @@
+To customize the existing row layout, implement the [onRowPrepared](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onRowPrepared) event handler. This handler allows you to customize the existing row layout after it is generated. If a customization is not immediately applied, use the [repaintRows(rowIndexes)](/Documentation/ApiReference/UI_Components/dxDataGrid/Methods/#repaintRowsrowIndexes) method to repaint DataGrid rows.
+
+#include common-demobutton with {
+    url: "https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/RowTemplate/"
+}
+
+The following example demonstrates how to use the [onRowPrepared](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onRowPrepared) function to change cell color conditionally. As a result, the DataGrid component paints the rows where velocity is higher than speed limit red.
+
+![Change Color Conditionally with onRowPrepared](/images/DataGrid/customize-appearance/onRowPrepared.png)
```

Comment:
I forgot to comment on that in the previous topic, but the screenshot data looks strange. First we have a **Fax Number** field and then a couple of fields related to Speed Limit. I'm not sure what kind of entity would have those kind of fields together. 

Another issue is that you compare fields called "Velocity" and "Speed Limit". The first sounds like something from a physics book. The second is something out of a street sign. I mean you can either use "Velocity"/"Maximum Velocity" if you talk about the movement of random objects (not cars), or you need to do "Speed"/"Speed Limit" if you do talk about cars. In other words, the term use should be consistent. 

Maybe you can replace the first field with a "Date / Time" field, and then use captions "Speed" and "Speed Limit". That would looks like a part of a speeding ticket report or something like that :-)